pkg/controller/keystoneapi/keystoneapi_controller.go

```diff
@@ -175,6 +177,27 @@ func (r *ReconcileKeystoneAPI) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{RequeueAfter: time.Second * 5}, err
 	}
 
+	// Create the DB Schema (unstructured so we don't explicitly import mariadb-operator code)
+	schemaObj := keystone.SchemaObject(instance)
+
+	foundSchema := &unstructured.Unstructured{}
+	foundSchema.SetGroupVersionKind(schemaObj.GroupVersionKind())
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: schemaObj.GetName(), Namespace: schemaObj.GetNamespace()}, foundSchema)
+	if err != nil && k8s_errors.IsNotFound(err) {
+		err := r.client.Create(context.TODO(), &schemaObj)
```

Comment:
I initially worried that this could be a race, but I think not because all created objects should have the same name, so only the first should succeed.

Incidentally, although we don't (currently, anyway) intend for this code to runnable concurrently, I would defensively always assume that it might run concurrently.

Reply:
only 1 Reconcile should run at a time. That said I think the code as written is probably robust enough to handle the case of concurrency in that if 2 threads somehow ran at the same time they would compete and 1 would pretty quickly throw a Reconcile Error and thus retry again. Essentially the way we are writing operators is always in a "make it so" fashion. If there is an error you log it, and retry accordingly hopefully eventually getting to the state you wanted.